test/Microsoft.Extensions.Http.Polly.Test/PolicyHttpMessageHandlerTest.cs

```diff
@@ -103,8 +97,65 @@ public class PolicyHttpMessageHandlerTest
             Assert.Same(expected, response);
         }
 
+        [Fact]
+        public async Task SendAsync_NoContextSet_CreatesNewContext()
+        {
+            // Arrange
+            var policy = Policy.TimeoutAsync<HttpResponseMessage>(TimeSpan.FromSeconds(10));
+            var handler = new TestPolicyHttpMessageHandler(policy);
+
+            Context context = null;
+            var expected = new HttpResponseMessage();
+            handler.OnSendAsync = (req, c, ct) =>
+            {
+                context = c;
+
+                Assert.NotNull(context);
+                Assert.Same(context, req.GetPollyContext());
+                return expected;
+            };
+
+            var request = new HttpRequestMessage();
+
+            // Act
+            var response = await handler.SendAsync(request, CancellationToken.None);
+
+            // Assert
+            Assert.Same(context, request.GetPollyContext()); // We don't clean up the context
+            Assert.Same(expected, response);
+        }
+
+        [Fact]
+        public async Task SendAsync_ExistingContext_ReusesContext()
```

Comment:
Test as is does not check that the correct context is passed to SendAsync?  SUT could push any other context onto `HttpRequestMessage` before calling `SendAsync(...)`, pop it off again after, and the test would still pass?  Could strengthen by tracking `Context contextPassedToSendAsync;`

Reply:
Sure, this got missed somehow.